test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,149 @@ func testManualMeteringInstall(
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	namespace, error := createNamespace(ctx)
+	if error != nil {
+		return error
+	}
+
+	//decode and create yamls except for persistentvolume
+
+	type nfsManifestItem struct {
+		resourceType interface{}
+		resourceFile string
+		actions      []string
+	}
+	type nfsServer struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsService struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsStorageclass struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+	var nfsManifestItems = map[string]nfsManifestItem{
+		"nfsServer":
+		{
+
+			resourceType: nfsServer{},
+			resourceFile: "/test/testhelpers/nfs_server.yaml",
+			actions:      []string{"CREATE"},
+		},
+		"nfsService":
+		{
+			resourceType: nfsService{},
+			resourceFile: "/test/testhelpers/nfs_service.yaml",
+			actions:      []string{"CREATE", "GET"},
+		},
+		"nfsStorageclass":
+		{
+			resourceType: nfsStorageclass{},
+			resourceFile: "/test/testhelpers/nfs_storageclass.yaml",
+			actions:      []string{"CREATE"},
+		},
+	}
+
+	for _, manifest := range nfsManifestItems {
+
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsServer"].resourceFile, nfsManifestItems["nfsServer"].resourceType)
+		err = deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsService"].resourceType)
+		err = deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsStorageclass"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceType)
+		if err != nil {
+			return fmt.Errorf("failed to decode the %s YAML service manifest: %v", err, manifest.resourceFile)
+		}
+
+		for _, action := range manifest.actions {
+			if action == "CREATE" {
+				ctx.Client.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"], nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+				_, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
+
+	//  - Wait for service to populate spec.ClusterIP (poll.Wait functionality)
+	// poll has a closure that polls for the server IP to get populated
+	var service, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+	// if theres an error don't poll
+	if err != nil {
+		err := wait.Poll(3*time.Second, 5*time.Minute, func() (done bool, err error) {
+			service, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+			return false, err
+		}
+		// clusterIP is populated, stop polling
+		return service.Spec.ClusterIP != "", nil
```

Comment:
You're returning this conditional outside of the wait.Poll conditional function.

Reply:
Fixed